.github/workflows/deploy-instance.yml

```diff
@@ -0,0 +1,57 @@
+name: Deploy instance
+
+on:
+  pull_request:
+    types: [opened, reopened, labeled, synchronize]
+
+# Cancel previous runs of the same workflow and PR number or branch/tag
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  deploy-instance:
+    if: contains(github.event.pull_request.labels.*.name, 'deploy')
+    runs-on: ubuntu-latest
+    env:
+      INSTANCE_HOST: 'https://dev.im.dhis2.org'
+      INSTANCE_NAME: pr-${{ github.event.pull_request.number }}
+    steps:
+      - name: Wait for API tests
+        uses: lewagon/wait-on-check-action@v1.3.1
+        with:
+          ref: ${{ github.head_ref }}
+          check-name: 'api-test'
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/checkout@v4
+        with:
+          repository: dhis2-sre/im-manager
+          sparse-checkout: scripts/instances
+
+      - name: Install HTTPie
+        run: python -m pip install httpie
+
+      - name: Deploy DHIS2 instance
+        working-directory: scripts/instances
+        env:
+          HTTP: https --check-status
+          USER_EMAIL: ${{ secrets.IM_BOT_EMAIL }}
+          PASSWORD: ${{ secrets.IM_BOT_PASSWORD }}
+          IM_HOST: 'https://api.im.dhis2.org'
+          IMAGE_REPOSITORY: core-pr
+          IMAGE_TAG: ${{ github.event.pull_request.number }}
+          IMAGE_PULL_POLICY: Always
+          DATABASE_ID: sl-sierra-leone-dev-sql-gz
+        run: ./findByName.sh dev $INSTANCE_NAME && ./restart.sh dev $INSTANCE_NAME || ./deploy-dhis2.sh dev $INSTANCE_NAME
+
+      - name: Wait for instance
+        run: timeout 300 bash -c 'while [[ "$(curl -fsSL -o /dev/null -w %{http_code} $INSTANCE_HOST/$INSTANCE_NAME)" != "200" ]]; do sleep 5; done'
+
+      - name: Generate analytics
+        run: curl -X POST -u 'admin:district' "$INSTANCE_HOST/$INSTANCE_NAME/api/resourceTables/analytics"
```

Comment:
could we add
`-d 'executeTei=true'`
to this command, to allow the generation of Cross Program Linelist analytics data, which wouldn't be generated otherwise?

Reply:
Sure, yeah. I'm just not sure if it could turn out to be a problem for any use case?

Reply:
It shouldn't, it's just an extra data export which is probably the fastest among all others.
It will "just" affect the disk space used.
BTW the PR is already good enough to me and what we're discussing might be addressed in another place and in other PRs.
I think we could make all analytics conditional on label like
- `deploy` just deploy the branch
- `deploy-analytics` deploy and runs analytics
- `deploy-analytics-cpl` deploy, runs analytics with `exportTei=true` flag

not sure if we could do that

Reply:
We could, but let's try to "keep it simple" and go with generating analytics with the extra Cross Program Linelist data. If any issues come out of it, we can always fix it.